wdl-ast/src/experimental/v1/expr.rs

```diff
@@ -0,0 +1,4922 @@
+                .value()
+                .unwrap(),
+            1234.1234
+        );
+
+        // Fourth declaration
+        assert_eq!(decls[3].ty().to_string(), "Float");
+        assert_eq!(decls[3].name().as_str(), "d");
+        assert_relative_eq!(
+            decls[3]
+                .expr()
+                .unwrap_literal()
+                .unwrap_float()
+                .value()
+                .unwrap(),
+            123e+123
+        );
+
+        // Fifth declaration
+        assert_eq!(decls[4].ty().to_string(), "Float");
+        assert_eq!(decls[4].name().as_str(), "e");
+        assert_relative_eq!(
+            decls[4]
+                .expr()
+                .unwrap_literal()
+                .unwrap_float()
+                .value()
+                .unwrap(),
+            0.1234
+        );
+
+        // Sixth declaration
+        assert_eq!(decls[5].ty().to_string(), "Float");
+        assert_eq!(decls[5].name().as_str(), "f");
+        assert_relative_eq!(
+            decls[5]
+                .expr()
+                .unwrap_literal()
+                .unwrap_float()
+                .value()
+                .unwrap(),
+            10.0
+        );
+
+        // Seventh declaration
+        assert_eq!(decls[6].ty().to_string(), "Float");
+        assert_eq!(decls[6].name().as_str(), "g");
+        assert_relative_eq!(
+            decls[6]
+                .expr()
+                .unwrap_literal()
+                .unwrap_float()
+                .value()
+                .unwrap(),
+            0.2
+        );
+
+        // Eighth declaration
+        assert_eq!(decls[7].ty().to_string(), "Float");
+        assert_eq!(decls[7].name().as_str(), "h");
+        assert!(
+            decls[7]
+                .expr()
+                .unwrap_literal()
+                .unwrap_float()
+                .value()
+                .is_none()
+        );
+
+        // Use a visitor to visit all the literal floats in the tree
+        struct MyVisitor(Vec<f64>);
+
+        impl Visitor for MyVisitor {
+            type State = ();
+
+            fn expr(&mut self, _: &mut Self::State, reason: VisitReason, expr: &Expr) {
+                if reason != VisitReason::Enter {
+                    return;
+                }
+
+                if let Expr::Literal(LiteralExpr::Float(f)) = expr {
+                    if let Some(f) = f.value() {
+                        self.0.push(f);
+                    }
+                }
+            }
+        }
+
+        let mut visitor = MyVisitor(Vec::new());
+        ast.visit(&mut (), &mut visitor);
+        assert_relative_eq!(
+            visitor.0.as_slice(),
+            [0.0, 0.0, 1234.1234, 123e123, 0.1234, 10.0, 0.2].as_slice()
+        );
+    }
+
+    #[test]
+    fn literal_string() {
+        let parse = Document::parse(
+            r#"
+version 1.1
+
+task test {
+    String a = "hello"
+    String b = 'world'
+    String c = "Hello, ${name}!"
```

Comment:
Is it captured by the AST whether a placeholder uses `~` or `$`?
And related, should we have a lint for ensuring `~` is always used?

Reply:
I think a lint to ensure `~{}` is consistently used (even outside `command` blocks) is a good choice.

Reply:
Sounds good all around 👍

Reply:
Added a `Placeholder::has_tilde` convenience method.